internal/eliza/eliza.go

```diff
@@ -36,6 +37,31 @@ func Reply(input string) (string, bool) {
 	return lookupResponse(input), false
 }
 
+// GetIntroResponses returns a collection of introductory responses tailored to the given name.
+func GetIntroResponses(name string) []string {
+	intros := []string{}
+	for _, n := range introResponses {
+		intros = append(intros, fmt.Sprintf(n, name))
+	}
+	return intros
+}
+
+// GetRandomDetails returns a random amount of details about Eliza.
+func GetRandomDetails() []string {
+	details := elizaDetails
+	total := len(details)
+
+	responses := getRandomIntInRange(1, total)
+	rand.Shuffle(total, func(i, j int) { details[i], details[j] = details[j], details[i] })
```

Comment:
This is a race - we're mutating `elizaDetails`, which is package-global. The `details := elizaDetails` line above is only copying the slice header, which uses a pointer to reference the backing array - it's not a deep copy of the slice.

How about always returning just one detail from the slice? Then we don't need the shuffle and don't run into this problem.

Reply:
Much easier.  In fact, this removes a lot of the random code because there's already a `randomElementFrom`.